### PR TITLE
Add ability to set table name when adding a worksheet with datatable …

### PR DESCRIPTION
**Pull request by NickNack2020**
Fixes Issue #1712  - Add ability to set table name when adding a worksheet with datatable.

### CODE REVIEW

File: ClosedXML/Excel/XLWorkbook.cs [hunk 1]
```diff
@@ -857,11 +852,41 @@ public void AddWorksheet(IXLWorksheet worksheet)
             worksheet.CopyTo(this, worksheet.Name);
         }
 
+        /// <summary>
+        /// Add a worksheet with a table at Cell(row:1, column:1). The datatable's name is used for the
+        /// worksheet name and the table name.
+        /// </summary>
+        /// <param name="dataTable">Datatable to insert</param>
+        /// <returns>Inserted Worksheet</returns>
+        public IXLWorksheet AddWorksheet(DataTable dataTable)
```
Comment (jahav) [inline ClosedXML/Excel/XLWorkbook.cs:855]: Please place comments to the interface, not an implementation. That way, users will see them through IntelliSense.

File: ClosedXML/Excel/XLWorksheet.cs [hunk 2]
```diff
@@ -1551,25 +1551,6 @@ private void CheckRangeNotOverlappingOtherEntities(XLRange range)
                 throw new InvalidOperationException($"The range {range.RangeAddress.ToStringRelative(includeSheet: true)} overlaps with the worksheet's autofilter.");
         }
 
-        private string GetNewTableName(string baseName)
```
Comment (NickNack2020) [inline ClosedXML/Excel/XLWorksheet.cs:1554]: Refactored out into a reusable class - See TableNameGenerator

File: ClosedXML/Excel/XLWorksheets.cs [hunk 3]
```diff
@@ -160,9 +160,14 @@ public IXLWorksheet Add(DataTable dataTable)
         }
 
         public IXLWorksheet Add(DataTable dataTable, String sheetName)
+        {
+            return Add(dataTable, sheetName, TableNameGenerator.GetNewTableName(_workbook));
```
Comment (NickNack2020) [inline ClosedXML/Excel/XLWorksheets.cs:164]: It uses the new table name generator class and provides a generated table name, as suggested in the ticket.

File: ClosedXML/Excel/XLWorkbook.cs (hunk 1 above)
Comment (NickNack2020) [reply ClosedXML/Excel/XLWorkbook.cs:855]: done

**Comment by NickNack2020** [comment]
@jahav I believe I made the suggested changes. Let me know if there is anything else.

**Comment by jahav** [comment]
Resolves #1712

**Pull request merged**